composer/algorithms/weight_standardization/README.md

```diff
@@ -0,0 +1,112 @@
+my_train_dataloader = DataLoader(RandomImageDataset(), batch_size=2)
+my_eval_dataloader = DataLoader(RandomImageDataset(), batch_size=2)
+```
+-->
+<!--pytest-codeblocks:cont-->
+```python
+# Instantiate the Weight Standardization algorithm and pass it into the Trainer
+# The trainer will automatically run it at the appropriate point in the training loop
+
+from composer.algorithms import WeightStandardization
+from composer.trainer import Trainer
+
+# Train model
+
+# Only ignore last layers if the classification layer(s) are convolutions
+weight_standardization = WeightStandardization(n_last_layers_ignore=0)
+
+trainer = Trainer(
+    model=cnn_composer_model,
+    train_dataloader=my_train_dataloader,
+    eval_dataloader=my_eval_dataloader,
+    max_duration='1ep',
+    algorithms=[weight_standardization]
+)
+
+trainer.fit()
+```
+
+### Implementation Details
+
+Weight standardization is implemented using [PyTorch's parametrization package](https://pytorch.org/tutorials/intermediate/parametrizations.html) to reparametrize convolutional weights to have zero mean and unit standard deviation across the input channel and kernel dimensions. This standardization is computed on each forward pass during training.
+
+The parameter `n_last_layers_ignore` specifies how many layers at the end of the network to ignore when applying weight standardization. This is essential if the classification layer(s) are convolutional since weight standardization should not be applied to the classification layer(s). Before applying weight standardization, the model is symbolically traced with [PyTorch's torch.fx](https://pytorch.org/docs/stable/fx.html). The symbolic trace provides the model's modules in the order they are executed instead of the order they are defined. Depending on the model, the symbolic trace could fail, resulting in a warning and potentially incorrect behavior when using `n_last_layers_ignore`.
+
+## Suggested Hyperparameters
+
+We found the best performance resulted from setting `n_last_layers_ignore` equal to the number of classification layers that are convolutional. For example, we set `n_last_layers_ignore=0` for training ResNet-50 on ImageNet since the classification is a single linear layer and we set `n_last_layers_ignore=15` for training DeepLabv3+ on ADE20k since the DeepLabv3+ classification head consists of 15 convolution layers.
+
+## Technical Details
+
+For ResNet-50 trained on ImageNet, we measured a +0.08% improvement when using weight standardization. For DeepLabv3+ with a ResNet-101 backbone trained on ADE20k, we measured a +0.7 mIoU improvement when using `n_last_layers_ignore=15`. In addition to the improvements at the end of training, we observed larger improvements early in training and sometimes a decrease in training loss. This suggests the potential for weight standardized models to be trained with more regularization. We have only tested the performance improvement from this method on ResNet architectures.
```

Comment:
Do we see this improvement when using pretrained weights, or only when training from scratch?

Reply:
Only training from scratch since I haven't tried using pre-trained weights trained with weight standardization. I will update the doc!